rpc/transaction.go

```diff
@@ -210,7 +210,7 @@ type Transaction struct {
 	ContractAddressSalt   *felt.Felt                   `json:"contract_address_salt,omitempty" validate:"required_if=Type DEPLOY,required_if=Type DEPLOY_ACCOUNT"`
 	ClassHash             *felt.Felt                   `json:"class_hash,omitempty" validate:"required_if=Type DEPLOY,required_if=Type DEPLOY_ACCOUNT"`
 	ConstructorCallData   *[]*felt.Felt                `json:"constructor_calldata,omitempty" validate:"required_if=Type DEPLOY,required_if=Type DEPLOY_ACCOUNT"`
-	SenderAddress         *felt.Felt                   `json:"sender_address,omitempty" validate:"required_if=Type DECLARE,required_if=Type INVOKE Version 0x1"`
+	SenderAddress         *felt.Felt                   `json:"sender_address,omitempty" validate:"required_if=Type DECLARE,required_if=Type INVOKE Version 0x1,required_if=Version 0x3"`
```

Comment:
I think it will require all tx-s to have this field if version = 3, which shouldn't be the case for https://docs.starknet.io/architecture-and-concepts/network-architecture/transactions/#deploy_account_transaction

Reply:
Yup, forgot to constrain to invoke type

Invoke v0 - not required
Invoke v1 - required
Invoke v3 - required
Declare v1 -required
Declare v2 -required
Declare v3 - required
Deploy v1 -  not required
Deploy v3 - not required